### PR TITLE
This is for issue #68

### PR DESCRIPTION
**Pull request by night-day**
Created a new utiility js for creating the TCG affiliate link

Updated the CardModal to use new funciton

Added middle click check via OnAuxClick on the ListView and open TCG link in new window

### CODE REVIEW

File: src/util/Affiliate.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    } else {
+      tcgplayerLink += `productcatalog/product/show?ProductName=${card.details.name}`;
+    }
+    tcgplayerLink += '&partner=CubeCobra&utm_campaign=affiliate&utm_medium=CubeCobra&utm_source=CubeCobra';
+  
+     return tcgplayerLink;
+}
+
+  
+export default { getTCGLink };
```
Comment (phulin) [inline src/util/Affiliate.js]: Can you change the indentation in this file to 2 spaces and remove the empty line at the start?

File: src/components/AutocardListItem.js [hunk 2]
```diff
@@ -25,6 +28,7 @@ const AutocardListItem = ({ card, noCardModal, children }) => {
                 <a
                   href={noCardModal ? undefined : '#'}
                   className="d-block w-100"
+                  onAuxClick={noCardModal ? undefined : e => { e.preventDefault(); if (e.button == 1) { window.open(Affiliate.getTCGLink(card)); }}}
```
Comment (phulin) [inline src/components/AutocardListItem.js]: This handler is complicated enough that it should probably be broken out into a separate function.

File: src/util/Affiliate.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+
+function getTCGLink(card) {
```
Comment (phulin) [inline src/util/Affiliate.js]: Might as well export this function as a named export, too.
```suggestion
export function getTCGLink(card) {
```

File: src/components/AutocardListItem.js (hunk 2 above)
Comment (night-day) [reply src/components/AutocardListItem.js]: Thanks for the suggestion, where should I define the function?  in that same js?

**Comment by night-day** [comment]
I updated and fixed based on phulin comments hoping it worked out

**Comment by purxiz** [comment]
References #68

**Pull request merged**